database_observability: make collectors configurable (#2530)
grafana/alloy: docs/sources/reference/components/database_observability/database_observability.mysql.md

You can use the following arguments with `database_observability.mysql`:

 Name                 | Type                 | Description                                                   | Default | Required 
----------------------|----------------------|---------------------------------------------------------------|---------|----------
 `data_source_name`   | `secret`             | [Data Source Name][] for the MySQL server to connect to.      |         | yes      
 `forward_to`         | `list(LogsReceiver)` | Where to forward log entries after processing.                |         | yes      

>  `collect_interval`   | `duration`           | How frequently to collect information from database.          | `"10s"` | no       
>  `enable_collectors`  | `list(string)`       | A list of [collectors][] to enable on top of the default set. |         | no       
>  `disable_collectors` | `list(string)`       | A list of [collectors][] to disable from the default set.     |         | no       

Reviewer: ```suggestion
 `collect_interval`   | `duration`           | How frequently to collect information from database.          | `"10s"` | no       
 `disable_collectors` | `list(string)`       | A list of [collectors][] to disable from the default set.     |         | no       
  `enable_collectors`  | `list(string)`       | A list of [collectors][] to enable on top of the default set. |         | no       
```

Just alphabetizing the "not required" arg in the table.

You can use the following arguments with `database_observability.mysql`:

>  Name                 | Type                 | Description                                                   | Default | Required 

Author: Neat preview feature. See changes here https://deploy-preview-alloy-2530-zb444pucvq-vp.a.run.app/docs/alloy/latest/reference/components/database_observability/database_observability.mysql/

Reviewer 2: Nice!

----------------------|----------------------|---------------------------------------------------------------|---------|----------
 `data_source_name`   | `secret`             | [Data Source Name][] for the MySQL server to connect to.      |         | yes      
 `forward_to`         | `list(LogsReceiver)` | Where to forward log entries after processing.                |         | yes      
 `collect_interval`   | `duration`           | How frequently to collect information from database.          | `"10s"` | no       
 `enable_collectors`  | `list(string)`       | A list of [collectors][] to enable on top of the default set. |         | no       
 `disable_collectors` | `list(string)`       | A list of [collectors][] to disable from the default set.     |         | no       

[collectors]: #supported-collectors

## Blocks

The `database_observability.mysql` component doesn't support any blocks. You can configure this component with arguments.

> ### Supported Collectors
> 
> The full list of supported collectors is:
> 
> | Name         | Description                                           | Enabled by default |
> |--------------|-------------------------------------------------------|--------------------|
> | query_sample | Collect query samples.                                | yes                |
> | schema_table | Collect schemas and tables from `information_schema`. | yes                |
> 
> 

Reviewer: ```suggestion
The following collectors are enabled by default:

* `query_sample`: Collect query samples.
* `schema_table`: Collect schemas and tables from `information_schema`.

```

I would suggest this is better represented as a bullet list.

I would also drop the heading and just move it up under the table around line 35-36.

> 
> [collectors]: #supported-collectors
> 
> ## Blocks
> 
> The `database_observability.mysql` component doesn't support any blocks. You can configure this component with arguments.
> 
> ### Supported Collectors
> 
> The full list of supported collectors is:
> 
> | Name         | Description                                           | Enabled by default |
> |--------------|-------------------------------------------------------|--------------------|
> | query_sample | Collect query samples.                                | yes                |
> | schema_table | Collect schemas and tables from `information_schema`. | yes                |
> 
> 

Reviewer: ```suggestion

The following collectors are enabled by default:

* `query_sample`: Collect query samples.
* `schema_table`: Collect schemas and tables from `information_schema`.

[collectors]: #supported-collectors

## Blocks

The `database_observability.mysql` component doesn't support any blocks. You can configure this component with arguments.

```

This info is probably better as a bullet list, and should be below the Args table, not a child of `Blocks`.

Author: Thanks for reviewing, @clayton-cornell! I agree it was a mistake to have it as a child of Blocks.
I was trying to model the page a little bit like the mysql exporter in prometheus [here](https://grafana.com/docs/alloy/latest/reference/components/prometheus/prometheus.exporter.mysql/).

 What do you think about the following:
* keeping it as a table to match the mysql exporter docs and because we will add more collectors in the future.
* Supported Collectors' is correctly moved to a child of Args

Author: No worries, perfect, thank you!
